autogpt_platform/backend/backend/blocks/screenshotone.py

```diff
@@ -0,0 +1,166 @@
+from base64 import b64encode
+from typing import Literal
+
+from pydantic import SecretStr
+
+from backend.data.block import Block, BlockCategory, BlockOutput, BlockSchema
+from backend.data.model import (
+    APIKeyCredentials,
+    CredentialsField,
+    CredentialsMetaInput,
+    SchemaField,
+)
+from backend.integrations.providers import ProviderName
+from backend.util.file import MediaFile, store_media_file
+from backend.util.request import Requests
+
+
+class ScreenshotWebPageBlock(Block):
+    """Block for taking screenshots using ScreenshotOne API"""
+
+    class Input(BlockSchema):
+        credentials: CredentialsMetaInput[
+            Literal[ProviderName.SCREENSHOTONE], Literal["api_key"]
+        ] = CredentialsField(description="The ScreenshotOne API key")
+        url: str = SchemaField(
+            description="URL of the website to screenshot",
+            placeholder="https://example.com",
+        )
+        viewport_width: int = SchemaField(
+            description="Width of the viewport in pixels", default=1920
+        )
+        viewport_height: int = SchemaField(
+            description="Height of the viewport in pixels", default=1080
+        )
+        full_page: bool = SchemaField(
+            description="Whether to capture the full page length", default=False
+        )
+        format: str = SchemaField(
+            description="Output format (png, jpeg, webp)", default="png"
+        )
+        block_ads: bool = SchemaField(description="Whether to block ads", default=True)
+        block_cookie_banners: bool = SchemaField(
+            description="Whether to block cookie banners", default=True
+        )
+        block_chats: bool = SchemaField(
```

Comment:
Should we add a Boolean option whether to return the image (base64) or return the Image file (relative path) to be passed around? We can make it non advanced default to return path. And user can toggle it if they want to see it on the builder.

Reply:
We should probably show on ui under both conditions to be honest